tools/query_impc/impc_tool.xml

```diff
@@ -1,313 +1,311 @@
+                        <expand macro="selectSeparator" />
+                    </when>
+                </conditional>
+                <param name="head" type="boolean" checked="true" truevalue="True" falsevalue="False" label="Choose if include the header in the output" help="The default value is True."/>
+            </when>
+            <when value="7">
+                <param name="input" type="boolean" checked="true" truevalue="True" falsevalue="False" label="Include genes without identified phenotypes?" help="Choose if include in the output table also those genes that have no registred phenotypes. By default they are excluded."/>
+                <param name="head" type="boolean" checked="true" truevalue="True" falsevalue="False" label="Choose if include the header in the output. The default value is True."/>
+                <expand macro="outputType" />
+            </when>
+            <when value="8">
+                <conditional name="inp_q">
+                    <expand macro="inputType" />
+                    <when value="str">
+                        <param name="input" type="text" label="Input pipeline" help="Enter a IMPReSS pipeline id"/>
+                    </when>
+                    <when value="txt">
+                        <param name="input" type="data" format="data,tabular,txt" label="Input file" help="Enter a txt file with an IMPReSS pipeline id"/>
+                        <expand macro="selectSeparator" />
+                    </when>
+                </conditional>
+                <param name="head" type="boolean" checked="true" truevalue="True" falsevalue="False" label="Choose if include the header in the output. The default value is True."/>
+                <expand macro="outputType" />
+            </when>
+            <when value="9">
+                <conditional name="inp_q">
+                    <expand macro="inputType" />
+                    <when value="str">
+                        <param name="input" type="text" label="Input id:" help="Enter a top level phenotype category id"/>
+                    </when>
+                    <when value="txt">
+                        <param name="input" type="data" format="data,tabular,txt" label="Input file" help="Enter a txt file with a top level phenotype category id"/>
+                        <expand macro="selectSeparator" />
+                    </when>
+                </conditional>
+                <param name="head" type="boolean" checked="true" truevalue="True" falsevalue="False" label="Choose if include the header in the output. The default value is True."/>
+                <expand macro="outputType" />
+            </when>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data format="tabular" name="output" label="${tool.name} query n° $query_type.selector"/>  
+    </outputs>
+    <tests>
+        <test>
```

Comment:
I think a few more tests would be nice, especially more combinations of conditionals.

Please also note that an explicit way of specifying there the conditionals is preferred. E.g. here: https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-tests-test-conditional

Reply:
Thank you @bgruening for your input. Just a clarification since it is not really clear to me even through examples: I have to use conditional if I'm making two different tests on the same query so I can merge them in a single test, or should I use it to have just a single test for more queries?

Reply:
Hi @bgruening , I added some more examples but I don't understand how to use those conditional in my tests.

Reply:
What I meant is please use this more explicit syntax to refer to a conditional (and sections):

https://github.com/galaxyproject/tools-iuc/blob/9b235992a96145b852de7a6b59bac3a33dcbbcc8/tools/circos/macros_tests.xml#L118

Really just use `<conitional name="foo">`. That makes it more obvious.

In addition more tests are always good :)

Reply:
@bgruening  I did like this but in that way only the two tests for query 1 are working:
<test>
   <conditional name="inp_q">
    <param name="input" value="test_query_2.txt"/>
    <param name="selector" value="2"/>
    <param name="head" value="True"/>
    <param name="inp_sel" value="txt"/>
    <param name="sep" value="t"/>
    <param name="g_out" value="mgi"/>
   </conditional>
   <output name="output" file="test_output_2.tabular"/>
  </test>

Reply:
Mh, that would be strange. Can you commit it merge everything and than I can have a look?

Reply:
Just didi it, but I think that there's a problem with the merging

Reply:
Ah I see. You could have just merged this PR here. Its a PR to your PR ... I know confusing :) Sorry.

Reply:
Yes sorry it is not very clear what's the issue now. Do I need to do something from my side?

Reply:
@bgruening  I followed the steps at the end of the page but the issue is still there, the files are loaded to git but the checks are not triggered

Reply:
@bgruening I'm not able to solve this merging problem. I've restored the old file version and the problem was solved, but as long as I make any change to it the error rises again.

Reply:
@bgruening ok I think I solved the merged issue and I uploaded the most recent xml file with all the changes